WcaOnRails/app/models/competition.rb

```diff
@@ -458,11 +465,82 @@ def in_progress?
   end
 
   def is_over?
-    end_date < Date.today
+    !!end_date && end_date < Date.today
+  end
+
+  def results_updated_at
+    Option.from_nilable(results.order('updated_at desc').pluck(:updated_at).first)
```

Comment:
I don't fully understand this Option class yet, so I might be missing something, but `results.order('updated_at desc').first.updated_at` seems simpler to me thank `pluck(...).first`.